src/components/editor/faultTree/Editor.tsx

```diff
@@ -181,23 +180,12 @@ const Editor = ({ setAppBarName }: DashboardTitleProps) => {
     event.preventDefault();
   };
 
-  const getScenariosWithHighestProbability = (list: FaultEventScenario[]): FaultEventScenario[] => {
-    if (list.length === 0) return [];
-
-    let highestProbability = list[0].probability;
-    let highestProbabilityScenarios: FaultEventScenario[] = [list[0]];
-
-    for (let i = 1; i < list.length; i++) {
-      const item = list[i];
-      if (item.probability > highestProbability) {
-        highestProbability = item.probability;
-        highestProbabilityScenarios = [item];
-      } else if (item.probability === highestProbability) {
-        highestProbabilityScenarios.push(item);
-      }
-    }
-
-    return highestProbabilityScenarios;
+  const getScenarioWithHighestProbability = (list) => {
+    if (list.length === 0) return null;
+    const result = list.reduce((maxItem, currentItem) => {
+      return currentItem.probability > maxItem.probability ? currentItem : maxItem;
+    });
+    return [result]
```

Comment:
Do we need to return array here?
I would expect to return `result` instead of `[result]`.

Reply:
Array removed.